Shopify/cli
# Allow partial deployments when sources are missing locally

### WHY are these changes introduced?

Fixes https://github.com/Shopify/internal-cli-foundations/issues/384

### WHAT is this pull request doing?

- Always ask for confirmation before deploying (user can provide `--force` to skip the confirmation prompt)
- Relax constraints so that the deploy can proceed even if an extension (or function) is missing locally
- Use new ink-powered prompts to display a summary for the user

### How to test your changes?

- create a new app: `pnpm create-app --path ~/Desktop`
- cd into app and create an extension `pnpm generate extension`
- open a new terminal side by side, go into the CLI folder and run `pnpm shopify app deploy --path ~/Desktop/new-app`
- you should see the confirmation prompt that you want to create a new extension
- type `n` to cancel
- run `pnpm shopify app deploy --path ~/Desktop/new-app --force` to deploy without any prompts
- in the app terminal, create a new extension `pnpm generate extension`
- in the CLI terminal,  and run `pnpm shopify app deploy --path ~/Desktop/new-app`
- you should see the prompt that you want to **create** a new extension and **update** an existing extension
- type 'y' to proceed
- in the app terminal, delete one of the extensions
- in the CLI terminal,  run `pnpm shopify app deploy --path ~/Desktop/new-app`
- you should see the prompt that you want to **update** an extension and that another extension is **missing locally**

### Post-release steps

- Coordinate with @shainaraskas for documentation changes

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
